### PR TITLE
Add NetCloud Section

### PR DESCRIPTION
**Pull request by stulle123**
Hi @vegantransistor,

This PR adds part 1 of the NetCloud write-up.

Would appreciate any feedback.

Cheers,

stulle123

### CODE REVIEW

File: netcloud/README.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+- [How to connect a router to NetCloud](https://customer.cradlepoint.com/s/article/NCM-Connecting-Devices)
+- [NetCloud API to query routers remotely](https://customer.cradlepoint.com/s/article/NCM-APIv2-Overview)
+  - [API query examples](https://developer.cradlepoint.com/)
+- [Enable remote administration](https://customer.cradlepoint.com/s/article/NCOS-How-to-Configure-Remote-Administration-on-a-Cradlepoint-Router)
+
+WPC Authentication / ECM:
+
+- https://accounts.cradlepointecm.com/
+  - Login with username and password
+  - Or login with `token_id` and `token_secret`
```
Comment (vegantransistor) [inline netcloud/README.md:22]: login over web interface only possible with username+password asaik.

File: netcloud/README.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    - Token stored in NOR dump
+    - `$ binwalk –extract nordump_pw0000.bin`
+    - `$ grep -rl "wpc.auth" .`
+- `ecm` command in the router's CLI
+  - Privileged sub-commands: `tri`, `ale`, `dif`
+- Connection
+  - Server: `stream.cradlepointecm.com`
+  - Port: `8001`
+  - Certificate: `/service_manager/services/wpcclient/stream.crt`
+- "Insecure Activation" (see `wpcclient.pyc`) probably only works if it's manually activated from the NetCloud side (see [use case]( https://customer.cradlepoint.com/s/article/How-To-Have-Cradlepoint-register-a-router-to-NetCloud-Manager-when-it-is-physically-not-accessible))
```
Comment (vegantransistor) [inline netcloud/README.md:32]: Insecure activation was also used to generate temporary credentials for brand-new devices (fixed by Cradlepoint after our report, fix tested by me).

File: netcloud/README.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+Read NetCloud token from a [patched router CLI](../README.md#flashing-the-nand-images):
```
Comment (vegantransistor) [inline netcloud/README.md:34]: make a subsection from this part?

File: netcloud/README.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+sudo sysctl -w net.ipv4.ip_forward=1
+sudo iptables -t nat -A POSTROUTING -o wlan0 -j MASQUERADE
+sudo iptables -A FORWARD -i eth0 -o wlan0 -j ACCEPT
+sudo iptables -A FORWARD -i wlan0 -o eth0 -m state --state RELATED,ESTABLISHED -j ACCEPT
+sudo iptables -t nat -A PREROUTING -i eth0 -p tcp --dport 8001 -j REDIRECT --to-port 8080
+sudo systemctl daemon-reload && sudo systemctl restart dhcpcd
+sudo service dnsmasq restart
+```
+
+2. Setup the router's WAN interface to use your proxy as the default gateway
```
Comment (vegantransistor) [inline netcloud/README.md:118]: maybe explain how

File: netcloud/scripts/README.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+Script for poking at the router registration process with NetCloud.
+
+Quick instructions:
+
+- Install the `netifaces` package: `$ pip install netifaces`
+- The script requires the `api` folder and the `stream.pyc` file
+  - Note: The `stream.pyc` file has been patched to work on a Mac. You may decompile it with [decompyle3](https://github.com/rocky/python-decompile3) and adapt it to your needs.
+- In the script, change the variables which have `Required` in the comment
+- Adjust the flags as needed to control the different registration stages. **Note**:
+  - `_INSECURE_ACTIVATION` only works for *new* devices that haven't been already registered with NetCloud.
```
Comment (vegantransistor) [inline netcloud/scripts/README.md:20]: actually it **worked**, this has been patched by Cradlepoint.

File: netcloud/scripts/test_netcloud_registration.py [hunk 6]
```diff
@@ -0,0 +1,394 @@
...
+import api.stream
+from api.base import RootException, Timeout, NotRegistered, Disabled, Unauthorized
+
+# Required: Add your NetCloud credentials (required for registering the router with NetCloud)
+_CID = "your-netcloud-user"
+_CPW = "your-netcloud-password"
+
+# Optional: Alternatively, use an existing NetCloud access token which you get from your rooted Cradlepoint device
+_TOKEN = {
+    "client_id": 844,
```
Comment (vegantransistor) [inline netcloud/scripts/test_netcloud_registration.py:20]: not sure about client_id, maybe change it to a variable.

**Review by vegantransistor — CHANGES_REQUESTED** [review]

File: netcloud/README.md (hunk 1 above)
Comment (stulle123) [reply netcloud/README.md:22]: Fixed.

File: netcloud/README.md (hunk 2 above)
Comment (stulle123) [reply netcloud/README.md:32]: Added.

File: netcloud/README.md (hunk 3 above)
Comment (stulle123) [reply netcloud/README.md:34]: Done.

File: netcloud/README.md (hunk 4 above)
Comment (stulle123) [reply netcloud/README.md:118]: Added.

File: netcloud/scripts/README.md (hunk 5 above)
Comment (stulle123) [reply netcloud/scripts/README.md:20]: Removed.

File: netcloud/scripts/test_netcloud_registration.py (hunk 6 above)
Comment (stulle123) [reply netcloud/scripts/test_netcloud_registration.py]: Done.

**Pull request merged**